### ISSUE TITLE
[BUG] java.io.EOFException with TiDB master

### ISSUE THREAD

**Issue by birdstorm** [pingcap/tispark#1379]
```
[2020-02-05T21:20:40.821Z] java.lang.RuntimeException: java.io.EOFException
[2020-02-05T21:20:40.821Z] 	at com.pingcap.tikv.codec.CodecDataInput.readUnsignedByte(CodecDataInput.java:173)
[2020-02-05T21:20:40.821Z] 	at com.pingcap.tikv.types.DataType.readIntLittleEndian(DataType.java:240)
[2020-02-05T21:20:40.821Z] 	at com.pingcap.tikv.types.DataType.decodeChunkColumn(DataType.java:263)
[2020-02-05T21:20:40.821Z] 	at com.pingcap.tikv.operation.iterator.CoprocessorIterator$2.next(CoprocessorIterator.java:150)
[2020-02-05T21:20:40.821Z] 	at com.pingcap.tikv.operation.iterator.CoprocessorIterator$2.next(CoprocessorIterator.java:121)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.tispark.TiRowRDD$$anon$1.next(TiRowRDD.scala:74)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.tispark.TiRowRDD$$anon$1.next(TiRowRDD.scala:51)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.coprocessorrdd_nextBatch_0$(Unknown Source)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$13$$anon$1.hasNext(WholeStageCodegenExec.scala:636)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:255)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:247)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
[2020-02-05T21:20:40.821Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:123)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
[2020-02-05T21:20:40.822Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2020-02-05T21:20:40.822Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2020-02-05T21:20:40.822Z] 	at java.lang.Thread.run(Thread.java:748)
[2020-02-05T21:20:40.822Z] Caused by: java.io.EOFException
[2020-02-05T21:20:40.822Z] 	at java.io.DataInputStream.readUnsignedByte(DataInputStream.java:290)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.codec.CodecDataInput.readUnsignedByte(CodecDataInput.java:171)
[2020-02-05T21:20:40.822Z] 	... 25 more
[2020-02-05T21:20:40.822Z] 20/02/06 05:20:40 WARN TaskSetManager: Lost task 0.0 in stage 309.0 (TID 309, localhost, executor driver): java.lang.RuntimeException: java.io.EOFException
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.codec.CodecDataInput.readUnsignedByte(CodecDataInput.java:173)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.types.DataType.readIntLittleEndian(DataType.java:240)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.types.DataType.decodeChunkColumn(DataType.java:263)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.operation.iterator.CoprocessorIterator$2.next(CoprocessorIterator.java:150)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.operation.iterator.CoprocessorIterator$2.next(CoprocessorIterator.java:121)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.tispark.TiRowRDD$$anon$1.next(TiRowRDD.scala:74)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.tispark.TiRowRDD$$anon$1.next(TiRowRDD.scala:51)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.coprocessorrdd_nextBatch_0$(Unknown Source)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$13$$anon$1.hasNext(WholeStageCodegenExec.scala:636)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:255)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.sql.execution.SparkPlan$$anonfun$2.apply(SparkPlan.scala:247)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:123)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
[2020-02-05T21:20:40.822Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
[2020-02-05T21:20:40.822Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2020-02-05T21:20:40.822Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2020-02-05T21:20:40.822Z] 	at java.lang.Thread.run(Thread.java:748)
[2020-02-05T21:20:40.822Z] Caused by: java.io.EOFException
[2020-02-05T21:20:40.822Z] 	at java.io.DataInputStream.readUnsignedByte(DataInputStream.java:290)
[2020-02-05T21:20:40.822Z] 	at com.pingcap.tikv.codec.CodecDataInput.readUnsignedByte(CodecDataInput.java:171)
[2020-02-05T21:20:40.822Z] 	... 25 more
```

TiDB master changed its codec logic.

**Comment by birdstorm** [comment]
Closed by #1382

**Comment by birdstorm** [comment]
Fixed by #1391
